tests/unit/store/createStore.ts

```diff
@@ -370,5 +380,80 @@ registerSuite({
 				assert.deepEqual(result, ids, 'Should have returned all deleted ids');
 			}).then(dfd.resolve);
 		}
+	},
+
+	'async storage': {
+		'async operation should not be done immediately.'(this: any) {
+			const{ dfd, store } = getStoreWithAsyncStorage(this, { put: 21 });
+
+			const start = Date.now();
+			store.add(createData()).then(dfd.callback(function(result: ItemType[]) {
+				const finish = Date.now();
+				assert.isAbove(finish - start, 19 );
+				dfd.resolve();
+			}));
+		},
+		'should complete initial add before subsequent operations'(this: any) {
+			const dfd = this.async(1000);
+			const asyncStorage = createAsyncStorage();
+			const store = createStore({
+				storage: asyncStorage,
+				data: createData()
+			});
+
+			store.get(['1', '2', '3']).then(dfd.callback(function(items: ItemType[]) {
```

Comment:
I agree. This is for consistency as well. Please see my above comments for details.